src/tools/dev/scripts/bv_support/bv_mpich.sh

```diff
@@ -22,13 +22,23 @@ function bv_mpich_depends_on
 
 function bv_mpich_info
 {
-    export MPICH_VERSION=${MPICH_VERSION:-"3.3.1"}
-    export MPICH_FILE=${MPICH_FILE:-"mpich-${MPICH_VERSION}.tar.gz"}
-    export MPICH_COMPATIBILITY_VERSION=${MPICH_COMPATIBILITY_VERSION:-"3.3"}
-    export MPICH_BUILD_DIR=${MPICH_BUILD_DIR:-"mpich-${MPICH_VERSION}"}
-    export MPICH_URL=${MPICH_URL:-http://www.mpich.org/static/tarballs/${MPICH_VERSION}}
-    export MPICH_MD5_CHECKSUM="9ed4cabd3fb86525427454381b25f6af"
-    export MPICH_SHA256_CHECKSUM="fe551ef29c8eea8978f679484441ed8bb1d943f6ad25b63c235d4b9243d551e5"
+    if [[ "$OPSYS" == "Darwin" ]]; then
```

Comment:
Updating mpich with newer libtool involves quite a bit. No easy way to just patch it. So, I created a new `third-party` asset named `mpich-3.3.1-libtool-2.4.6` which expands also to `mpich-3.3.1` (honestly should be harmless if its used on non-macOS systems too).

Reply:
This commit is needed for macOS 12.6. So I cherry picked it to #19293